src/persist-txn/src/operator.rs

```diff
@@ -263,12 +261,14 @@ where
         let mut read_data_to = empty_to;
         let mut output_progress_exclusive = T::minimum();
         loop {
-            // TODO(txn): Do something more specific when the input returns None
-            // (is shut down). I _think_ we'll always return before this happens
-            // because we'll get a Progress event with an empty frontier, and so
-            // we could assert on this, but I want to be more sure before making
-            // that change.
-            while let Some(event) = passthrough_input.next_mut().await {
+            loop {
+                // This only returns None when there are no more data left. Turn
```

Comment:
Should this still have the `TODO` without the `(txn)`?

Reply:
Heh, this was me attempting to address the TODO! @aljoscha this originally came from a PR comment from you, curious if you had anything else in particular in mind for this?

Reply:
If it works like this, which I think it should, we're good here :ok_hand:

Reply:
Cool! \o/